src/models/event.js

```diff
@@ -741,37 +741,34 @@ utils.extend(module.exports.MatrixEvent.prototype, {
     },
 
     /**
-     * Get whether the event is a relation event.
+     * Get whether the event is a relation event, and of a given type if `relType` is passed in.
+     *
+     * @param {string?} relType if given, checks that the relation is of the given type
      * @return {boolean}
      */
-    isRelation() {
+    isRelation(relType = undefined) {
         const content = this.getContent();
         const relation = content && content["m.relates_to"];
-        return relation && relation.rel_type && relation.event_id;
-    },
-
-    isReplacement() {
-        const content = this.getContent();
-        const relation = content && content["m.relates_to"];
-        return relation && relation.rel_type === "m.replace" && relation.event_id;
-    },
-
-    setReplacedEvent(replacedEvent) {
-        this._replacedEvent = replacedEvent;
+        return relation && relation.rel_type && relation.event_id &&
+            ((relType && relation.rel_type === relType) || !relType);
     },
 
-    getReplacedEvent() {
-        return this._replacedEvent;
+    /**
+     * Set an event that replaces the content of this event, through an m.replace relation.
+     *
+     * @param {MatrixEvent} newEvent the event with the replacing content.
+     */
+    makeReplaced(newEvent) {
+        this._replacingEvent = newEvent;
```

Comment:
Should this directly set `this.event.content`, rather than checking the replacing event during `getContent`? It depends whether you want to keep access to the unreplaced content...

Reply:
felt nicer to not modify the actual event ... I don't have any pref.

Reply:
Maybe it's nicer to not be doing Object.assign in getContent that way though, will have a look.

Reply:
Well, let's go with this for now, can adjust later.

Reply:
Did so in the end :+1: